File: Data Structures & Algorithms/same-binary-tree/submission-4.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    public boolean isSameTree(TreeNode p, TreeNode q) {
        if((p == null && q != null) || (p != null && q == null)){
            return false;
        }
        if(p == null && q == null){
            return true;
        }
        Queue<TreeNode> q1 = new LinkedList<>();
        Queue<TreeNode> q2 = new LinkedList<>();

        q1.offer(p);
        q2.offer(q);

        while(!q1.isEmpty() && !q2.isEmpty()){
            TreeNode currN1 = q1.remove();
            TreeNode currN2 = q2.remove();
            if(currN1 != null && currN2 != null && (currN1.val != currN2.val)){
                return false;
            }
            if((currN1.left == null && currN2.left != null)
                || (currN1.left != null && currN2.left == null)){
                    return false;
            }
            if((currN1.right == null && currN2.right != null)
                || (currN1.right != null && currN2.right == null)){
                    return false;
            }
            if(currN1.left != null){
                q1.offer(currN1.left);
            }
            if(currN2.left != null){
                q2.offer(currN2.left);
            }
            if(currN1.right != null){
                q1.offer(currN1.right);
            }
            if(currN2.right != null){
                q2.offer(currN2.right);
            }
        }
        return true;
    }
}

File: Data Structures & Algorithms/balanced-binary-tree/submission-4.js
/**
 * Definition for a binary tree node.
 * class TreeNode {
 *     constructor(val = 0, left = null, right = null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    /**
     * @param {TreeNode} root
     * @return {boolean}
     */
    isBalanced(root) {
        let height = true
        let posorder = function (node){
            if(node === null)return 0 ; 
           let leftheight =  posorder(node.left); 
            let rightheight = posorder(node.right); 
            if( Math.abs(leftheight - rightheight ) > 1 ){
                height = false
            }
             return  Math.max(leftheight, rightheight) + 1 ;
        }
        posorder(root)

       return height
    }
}
